src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -330,73 +330,75 @@ def list_role_assignment_change_logs(cmd, start_time=None, end_time=None):  # py
-                        entry['scope'] = payload['scope']
-                    if entry['scope']:
-                        index = entry['scope'].lower().find('/providers/microsoft.authorization')
-                        if index != -1:
-                            entry['scope'] = entry['scope'][:index]
-                        parts = list(filter(None, entry['scope'].split('/')))
-                        entry['scopeName'] = parts[-1]
-                        if len(parts) < 3:
-                            entry['scopeType'] = 'Subscription'
-                        elif len(parts) < 5:
-                            entry['scopeType'] = 'Resource group'
-                        else:
-                            entry['scopeType'] = 'Resource'
-
-                    # Look up the resource `name`, like b24988ac-6180-42a0-ab88-20f7382dd24c
-                    role_resource_name = payload['roleDefinitionId'].split('/')[-1]
-                    entry['roleDefinitionId'] = role_resource_name
-                    # In case the role definition has been deleted.
-                    entry['roleName'] = role_defs.get(role_resource_name, "N/A")
-
-            result.append(entry)
-
-    # Fill in logical user/sp names as guid principal-id not readable
-    principal_ids = {x['principalId'] for x in result if x['principalId']}
-    if principal_ids:
-        graph_client = _graph_client_factory(cmd.cli_ctx)
-        stubs = _get_object_stubs(graph_client, principal_ids)
-        principal_dics = {i.object_id: _get_displayable_name(i) for i in stubs}
-        if principal_dics:
-            for e in result:
-                e['principalName'] = principal_dics.get(e['principalId'], None)
+                    try:
+                        payload = json.loads(payload)
+                    except ValueError:
+                        pass
+                    if payload:
+                        if payload.get('properties') is None:
+                            continue
+                        payload = payload['properties']
+                        entry['principalId'] = payload['principalId']
+                        if not entry['scope']:
+                            entry['scope'] = payload['scope']
+                        if entry['scope']:
+                            index = entry['scope'].lower().find('/providers/microsoft.authorization')
+                            if index != -1:
+                                entry['scope'] = entry['scope'][:index]
+                            parts = list(filter(None, entry['scope'].split('/')))
+                            entry['scopeName'] = parts[-1]
+                            if len(parts) < 3:
+                                entry['scopeType'] = 'Subscription'
+                            elif len(parts) < 5:
+                                entry['scopeType'] = 'Resource group'
+                            else:
+                                entry['scopeType'] = 'Resource'
```

Comment:
Too many indents. Too long. Hard to read. Maybe you need to create some functions.

Reply:
True, but I intent not to change the code unless absolutely necessary.

Reply:
True, but I intend not to change the code unless absolutely necessary.